### PR TITLE
Streamline exercise

### PR DESCRIPTION
**Pull request by DillonN**

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -48,7 +50,7 @@
 	//      "mounts": [ "source=${env:HOME}${env:USERPROFILE}/.aspnet/https,target=/home/vscode/.aspnet/https,type=bind" ],
 
 	// Use 'postCreateCommand' to run commands after the container is created.
-	// "postCreateCommand": "dotnet restore",
+	"postCreateCommand": "dotnet restore",
```
Comment (DillonN) [inline .devcontainer/devcontainer.json:53]: This fixes most of the weird omnisharp issues we saw on first bootup

File: src/Life.Tests/BoardTest.cs [hunk 2]
```diff
@@ -6,25 +6,20 @@ namespace Life.Tests;
 
 public class BoardTest
 {
-    [Fact]
-    public void Should_Initialize_Board()
+    [Theory]
+    [MemberData(nameof(TestBoards.BoardScenarios), MemberType = typeof(TestBoards))]
```
Comment (DillonN) [inline src/Life.Tests/BoardTest.cs:10]: Reordered some things here to more logically follow the puzzle. Also parameterized the board tests so other test cases can be easily enabled

File: src/Life/Board.cs [hunk 3]
```diff
@@ -23,65 +23,13 @@ public void Evolve()
         throw new NotImplementedException();
     }
 
-    private int NextStateForCell(int i, int j)
```
Comment (DillonN) [inline src/Life/Board.cs:26]: I feel that adding these private methods is more part of the exercise. The public overrides at the bottom aren't used anywhere, so I would say either remove them or include them in tests

File: src/Life/Board.cs [hunk 4]
```diff
@@ -23,65 +23,13 @@ public void Evolve()
...
-
-    public override int GetHashCode()
-    {
-        throw new NotImplementedException();
-    }
-
     public override string ToString()
     {
-        throw new NotImplementedException();
+        return $"{nameof(State)}: {State}, {nameof(Rows)}: {Rows}, {nameof(Columns)}: {Columns}";
```
Comment (DillonN) [inline src/Life/Board.cs:28]: Not tested and more for convenience I think

File: src/Life/BoardExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
+namespace Life;
+
+public static class BoardExtensions
+{
+    public static void Print(this Board board)
```
Comment (DillonN) [inline src/Life/BoardExtensions.cs:5]: Separating the concern from the exercise

**Review by schweitzer-m — APPROVED** [review]

File: src/Life/Board.cs (hunk 3 above)
Comment (schweitzer-m) [reply src/Life/Board.cs:26]: I agree, if anyone is really struggling we can give this function names as an example to help get them on the right track.

**Pull request merged**